lib/codequest_pipes/pipe.rb

```diff
@@ -50,23 +56,36 @@ def self._check_interface(klass)
     private_class_method :_check_interface
 
     def self._required_context_elements
-      @required_context_elements ||= []
+      @required_context_elements ||= {}
     end
     private_class_method :_required_context_elements
 
     def self._provided_context_elements
-      @provided_context_elements ||= []
+      @provided_context_elements ||= {}
     end
     private_class_method :_provided_context_elements
 
     def self._validate_ctx(collection, ctx)
-      collection.each do |element|
-        next if ctx.respond_to?(element)
-        fail MissingContext, "context does not respond to '#{element}'"
+      collection.each do |element, cls|
```

Comment:
If `cls` is supposed to be short for `Class`, I believe convention has a handy `klass` word for it.

Reply:
👍